src/coprocessor/dag/expr/math.rs

```diff
@@ -0,0 +1,351 @@
+        let ctx = StatementContext::default();
+        for tt in tests {
+            let arg = datum_expr(tt.1);
+            let op = Expression::build(fncall_expr(tt.0, &[arg]), 0, &ctx).unwrap();
+            match tt.0 {
+                ScalarFuncSig::AbsInt | ScalarFuncSig::AbsUInt => {
+                    let got = op.eval_int(&ctx, &[]).unwrap_err();
+                    assert!(check_overflow(got).is_ok());
+                }
+                _ => unreachable!(),
+            }
+        }
+    }
+
+    #[test]
+    fn test_ceil() {
+        let tests = vec![
+            (ScalarFuncSig::CeilReal, Datum::F64(3.45), Datum::F64(4f64)),
+            (
+                ScalarFuncSig::CeilReal,
+                Datum::F64(-3.45),
+                Datum::F64(-3f64),
+            ),
+            (
+                ScalarFuncSig::CeilReal,
+                Datum::F64(f64::MAX),
+                Datum::F64(f64::MAX),
+            ),
+            (
+                ScalarFuncSig::CeilReal,
+                Datum::F64(f64::MIN),
+                Datum::F64(f64::MIN),
+            ),
+            (
+                ScalarFuncSig::CeilIntToInt,
+                Datum::I64(i64::MIN),
+                Datum::I64(i64::MIN),
+            ),
+            (
+                ScalarFuncSig::CeilIntToInt,
+                Datum::U64(u64::MAX),
+                Datum::U64(u64::MAX),
+            ),
+            (
+                ScalarFuncSig::CeilIntToDec,
+                Datum::I64(i64::MIN),
+                str2dec("-9223372036854775808"),
+            ),
+            (
+                ScalarFuncSig::CeilDecToInt,
+                str2dec("123.456"),
+                Datum::I64(124),
+            ),
+            (
+                ScalarFuncSig::CeilDecToInt,
+                str2dec("-123.456"),
+                Datum::I64(-123),
+            ),
+            (
+                ScalarFuncSig::CeilDecToDec,
+                str2dec("9223372036854775808"),
+                str2dec("9223372036854775808"),
+            ),
+        ];
+        let mut ctx = StatementContext::default();
+        ctx.ignore_truncate = true; // for ceil decimal to int.
+        for (sig, arg, exp) in tests {
+            let arg = datum_expr(arg);
+            let mut op = Expression::build(fncall_expr(sig, &[arg.clone()]), 0, &ctx).unwrap();
+            let exp = Expression::build(datum_expr(exp), 0, &ctx).unwrap();
+            if mysql::has_unsigned_flag(arg.get_field_type().get_flag()) {
+                op.mut_tp().set_flag(types::UNSIGNED_FLAG as u32);
+            }
+            match sig {
+                ScalarFuncSig::CeilReal => {
+                    let got = op.eval_real(&ctx, &[]).unwrap();
+                    let exp = exp.eval_real(&ctx, &[]).unwrap();
+                    assert_eq!(got, exp);
+                }
+                ScalarFuncSig::CeilIntToInt | ScalarFuncSig::CeilDecToInt => {
```

Comment:
Here I keep old code because `op.eval_int(...)` returns an `i64` with sign. If we put it into a `Datum`, We can't compare it with another `Datum::U64(number)`. @BusyJay